lib/pubsub/topic.js

```diff
@@ -87,25 +108,75 @@ Topic.formatName_ = function(projectId, name) {
  * @param {function=} callback - The callback function.
  *
  * @example
- * topic.publish('New message!', function(err) {});
+ * //-
+ * // You may publish a simple string message.
+ * //-
+ * var simpleMessage = 'new user';
+ * topic.publish(simpleMessage, function(err) {});
  *
- * topic.publish({
- *   user_id: 3,
+ * //-
+ * // Other times, you may need more information than a string can provide. Use
+ * // an object.
+ * //-
+ * var objectMessage = {
+ *   userId: 3,
  *   name: 'Stephen',
- *   message: 'Hello from me!'
- * }, function(err) {});
+ *   event: 'new user'
+ * };
+ * topic.publish(objectMessage, function(err) {});
+ *
+ * //-
+ * // You can also publish a batch of messages at once by supplying an array.
+ * //-
+ * topic.publish([ simpleMessage, objectMessage ], function(err) {});
  */
 Topic.prototype.publish = function(message, callback) {
   if (!message) {
     throw new Error('Cannot publish an empty message.');
   }
+
   callback = callback || util.noop;
-  if (!util.is(message, 'string') && !util.is(message, 'buffer')) {
-    message = JSON.stringify(message);
+
+  if (util.is(message, 'array')) {
+    this.publishBatch(message, callback);
+  } else {
+    this.publishRaw(Topic.formatMessage_(message), callback);
+  }
+};
+
+/**
+ * Publish a batch of messages at once.
+ *
+ * @param {array} messages - The array of messages.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * var messages = [
+ *   // The simple string format is accepted...
+ *   'user registered',
+ *
+ *   // ...as are raw message objects.
+ *   {
+ *     label: 'registration',
+ *     data: new Buffer('new user: stephen').toString('base64')
+ *   }
+ * ];
+ *
+ * topic.publishBatch(messages, function(err) {});
+ */
+Topic.prototype.publishBatch = function(messages, callback) {
```

Comment:
Wondering if we should make this a `_privateMethod` and just support `Topic#publish` which will accept a message or array of messages.